### PR TITLE
Automatic Switching to Dark Mode in Diagrams

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bug fix
- [x] Feature
- [ ] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

The diagram remained white, even in dark mode.

Issue Number: #2018

### What is the new behavior?

Via media queries the styling can be changed based on the color style of the user: light or dark mode.

Media queries can be defined according to the specs:

```css
@media (prefers-color-scheme = dark) { ... }
```

and via a more convenient shortcut:

```css
@media dark-mode { ... }
```

Both a `dark-mode` and `light-mode` can be used. Both those media queries apply to the screen only. For exporting those sections are skipped (e.i. `dark_mode is None`).

### CODE REVIEW

File: docs/style_sheets.md [hunk 1]
```diff
@@ -212,6 +212,32 @@ be provided to set the level of line wobbliness. 0.5 is default, 0.0 is a
...
+}
+
+/* Use a slightly grey background in the editor: */
+@media light-mode {
+  diagram {
+    background-color: #e1e1e1;
+  }
+}
+
+/* And antracit a slightly grey background in the editor: */
```
Comment (danyeaw) [inline docs/style_sheets.md:233]: I'm not sure what antracit means, is this a typo?

File: gaphor/ui/elementeditor.py [hunk 2]
```diff
@@ -288,7 +290,7 @@ def open(self, builder):
 
         self.style_sheet_buffer.set_language(
             self.lang_manager.get_language(
-                "css" if Gtk.get_major_version() == 3 else "gaphorcss"
+                "css"  # if Gtk.get_major_version() == 3 else "gaphorcss"
```
Comment (danyeaw) [inline gaphor/ui/elementeditor.py:293]: Is this supposed to be commented out?

File: docs/style_sheets.md (hunk 1 above)
Comment (amolenaar) [reply docs/style_sheets.md:233]: I suppose. I meant anthracite, the color :wink:

File: gaphor/ui/elementeditor.py (hunk 2 above)
Comment (amolenaar) [reply gaphor/ui/elementeditor.py:293]: Whoops!

**Pull request merged**